activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/JobExecutorIT.java

```diff
@@ -0,0 +1,387 @@
+                                                      .singleResult()
+                                                      .getId();
+        //when
+        for(int i=0; i<jobCount; i++)
+            runtimeService.createProcessInstanceBuilder()
+                          .processDefinitionId(processDefinitionId)
+                          .start();
+
+        // then
+        await("the async executions should complete and no more jobs should exist")
+                .untilAsserted(() -> {
+                    assertThat(runtimeService.createExecutionQuery()
+                                             .processDefinitionKey(ASYNC_TASK).count()).isEqualTo(0);
+                    
+                    assertThat(managementService.createJobQuery()
+                               .processDefinitionId(processDefinitionId)
+                               .count()).isEqualTo(0); 
+                });
+
+        assertThat(jobsCompleted.await(1, TimeUnit.MINUTES)).as("should complete all jobs")
+                                                            .isTrue();
+        // message handler is invoked
+        verify(jobMessageHandler, times(jobCount)).handleMessage(ArgumentMatchers.<Message<?>>any());
+    }
+    
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testCatchingTimerEvent() throws Exception {
+        CountDownLatch jobsCompleted = new CountDownLatch(1);
+        CountDownLatch timerScheduled = new CountDownLatch(1);
+        CountDownLatch timerFired = new CountDownLatch(1);
+
+        // Set the clock fixed
+        Date startTime = new Date();
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(timerScheduled),
+                                        ActivitiEventType.TIMER_SCHEDULED);
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(timerFired),
+                                        ActivitiEventType.TIMER_FIRED);
+
+        runtimeService.addEventListener(new CountDownLatchActvitiEventListener(jobsCompleted),
+                                        ActivitiEventType.JOB_EXECUTION_SUCCESS);
+
+        // when
+        ProcessInstance pi = runtimeService.startProcessInstanceByKey(INTERMEDIATE_TIMER_EVENT_EXAMPLE);
+
+        // then
+        assertThat(pi).isNotNull();
+        
+        await("the timer job should be created")
+            .untilAsserted(() -> {
+                assertThat(managementService.createTimerJobQuery()
+                                            .processInstanceId(pi.getId())
+                                            .count()).isEqualTo(1);
+            });
+
+        // After setting the clock to time '5 minutes and 5 seconds', the timer should fire
+        processEngineConfiguration.getClock().setCurrentTime(new Date(startTime.getTime() + ((5 * 60 * 1000) + 5000)));
+
+        // timer event has been scheduled
+        assertThat(timerScheduled.await(1, TimeUnit.MINUTES)).as("should schedule timer")
```

Comment:
Do we really need to wait that long?

Reply:
It takes time to schedule timer job especially in slow Travis. It will proceed as soon as latch is triggered maybe within 10 sec, so it is not a big deal to have 1 min timeout.

Reply:
Actually, I think the main reason for the test being failing in Travis is that the configuration defined in the setUp method (https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/188/files#diff-134ea6fb38fe95c3f6fa06d9ef418f38R109) is not being taken into account. I fixed the issue in Activiti core and it should be fixed here once this PR is merged
https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/326.

I checked locally, all the tests around timers are waiting between 8+ and 9+ seconds. The default timeout for awaitility is 10 seconds, so if Travis is slightly slower that's enough to make the test fail.